census/controllers/admin.js

```diff
@@ -66,10 +69,31 @@ let loadQuestionSets = function(req, res) {
     });
 };
 
+let buildIndexSite = function(req, res) {
+  /*
+  Generate and deploy an Index site for the site specified in the request.
+  */
+
+  // Set env for process to include DEBUG option, for more comprehensive
+  // logging.
+  const env = process.env;
+  // env.DEBUG = 'metalsmith-godi-*';
+
+  exec(`node index/generate.js ${res.locals.domain} -dsy ${res.locals.surveyYear}`,
```

Comment:
In theory, the value for `res.locals.domain` is checked before it reaches this action, but it wouldn't take many code changes to open this up to abuse.

Using `exec` with user-provided arguments does have the possibility of misuse. I've committed a change to use `execFile` instead which doesn't spawn a new shell, but instead calls the passed file directly.